src/Database/PostgreSQL/PQTypes/Internal/C/Types.hsc

```diff
@@ -236,6 +239,62 @@ instance Storable PGbytea where
 
 ----------------------------------------
 
+-- PGuuid C type is *char, but in actual expects the array
+-- to be 16 bytes. The bytes are encoded directly from the
+-- hex representation of the UUID, which is different from
+-- the default binary encoding from Data.UUID.
```

Comment:
Does system endianess have a play here?

Reply:
Not too sure. Do you know any way to verify and test that?

Reply:
Now I remember. Initially I used the system endianness to decode the `*char` as 4 `Word32`s. But it turned out libpqtypes or postgres is using _different_ endianness, so if we read out from `psql` as text it is decoded into the wrong hex string.